service/storage_service.cc

```diff
@@ -515,6 +520,8 @@ class topology_node_mutation_builder {
     topology_node_mutation_builder& set(const char* cell, const sstring& value);
     topology_node_mutation_builder& set(const char* cell, const raft::server_id& value);
     topology_node_mutation_builder& set(const char* cell, const std::unordered_set<dht::token>& value);
+    template<ConvertibleToSstring S>
+    topology_node_mutation_builder& set(const char* cell, const std::set<S>& value);
```

Comment:
std::convertible_to<sstring>

Reply:
Right, I forgot it exists...

Reply:
Actually, it's not the same. It appears that `std::string_view` is not `std::convertible_to<sstring>`. The overload does not match on `std::set<std::string_view>` and instead it matches on the first overload, which converts the set to `sstring` via `::format`...

It looks like the `sstring` constructor that accepts `std::string_view` is explicit, and `std::convertible_to` does not take it into account.

`std::constructible_from<sstring, S>` seems to do the job.